zenml/steps/trainer/pytorch_trainers/torch_ff_trainer.py

```diff
@@ -98,7 +101,45 @@ def __init__(self,
             last_activation=self.last_activation,
             input_units=self.input_units,
             output_units=self.output_units,
+            device = device,
             **kwargs)
+    
+    @staticmethod
+    def _assign_device(device:str):
+
+        def _choose_default_device():
```

Comment:
If `_assign_device` is  refactored then this can also be taken out as an inner function